src/routes/transactions/tests/preview.rs

```diff
@@ -0,0 +1,141 @@
+use std::time::Duration;
+
+use mockall::predicate::eq;
+use rocket::http::{ContentType, Header, Status};
+use rocket::local::asynchronous::Client;
+use serde_json::json;
+
+use crate::common::models::backend::chains::{
+    BlockExplorerUriTemplate, ChainInfo, GasPrice, NativeCurrency, RpcAuthentication, RpcUri, Theme,
+};
+use crate::config::contract_info_request_timeout;
+use crate::providers::info::*;
+use crate::routes::contracts::models::DataDecoderRequest;
+use crate::routes::transactions::handlers::preview::TransactionPreview;
+use crate::routes::transactions::tests::PREVIEW_RESPONSE;
+use crate::tests::main::setup_rocket;
+use crate::utils::errors::ApiResult;
+use crate::utils::http_client::{MockHttpClient, Request, Response};
+
+fn get_chain_info() -> ChainInfo {
+    ChainInfo {
+        recommended_master_copy_version: "1.1.1".to_string(),
+        transaction_service: "https://safe-transaction.mainnet.gnosis.io".to_string(),
+        vpc_transaction_service: "http://mainnet-safe-transaction-web.safe.svc.cluster.local"
+            .to_string(),
+        chain_id: "1".to_string(),
+        chain_name: "".to_string(),
+        short_name: "eth".to_string(),
+        l2: false,
+        description: "Random description".to_string(),
+        rpc_uri: RpcUri {
+            authentication: RpcAuthentication::ApiKeyPath,
+            value: "".to_string(),
+        },
+        safe_apps_rpc_uri: RpcUri {
+            authentication: RpcAuthentication::ApiKeyPath,
+            value: "".to_string(),
+        },
+        public_rpc_uri: RpcUri {
+            authentication: RpcAuthentication::ApiKeyPath,
+            value: "".to_string(),
+        },
+        block_explorer_uri_template: BlockExplorerUriTemplate {
+            address: "".to_string(),
+            tx_hash: "".to_string(),
+            api: "".to_string(),
+        },
+        native_currency: NativeCurrency {
+            name: "".to_string(),
+            symbol: "".to_string(),
+            decimals: 0,
+            logo_uri: "https://test.token.image.url".to_string(),
+        },
+        theme: Theme {
+            text_color: "#fff".to_string(),
+            background_color: "#000".to_string(),
+        },
+        ens_registry_address: None,
+        gas_price: vec![GasPrice::Fixed {
+            wei_value: "1000000".to_string(),
+        }],
+        disabled_wallets: vec![],
+        features: vec![],
+    }
+}
+
+#[rocket::async_test]
+async fn post_preview_success() {
```

Comment:
I would expect a couple more tests (e.g. what if a sub-request fails)

Reply:
Yes I agree. The test setup that we have didn't help and I wanted to provide something that was more in a reviewable state.

But I completely agree with you.